react/utils/genenarateUrl.js

```diff
@@ -0,0 +1,40 @@
+export const DEFAULT_WIDTH = 'auto'
+export const DEFAULT_HEIGHT = 'auto'
+export const MAX_WIDTH = 3000
+export const MAX_HEIGHT = 4000
+export const RESIZE_RATE = 1 / 2
+
+const baseUrlRegex = new RegExp(/.+ids\/(\d+)(?:-(\d+)-(\d+)|)\//)
+const sizeRegex = new RegExp(/-(\d+)-(\d+)/)
+
+export function cleanImageUrl(imageUrl) {
```

Comment:
Could you explain why is all this logic necessary?

The regexs and magic numbers makes this very hard for future improvements or bug fixes

Reply:
It's just to change the dimensions of the image URL. I did not saw another option at the time, do you know a better way to do that?

Reply:
The `result` part seems difficult to understand, and what the regex will match. A couple of line comments could help that. 

Also, what's the problem of replacing the matched string with the size regex in both scenarios of the `if` clause?